controllers/dspipeline_params.go

```diff
@@ -122,28 +122,71 @@ func passwordGen(n int) string {
+}
+
+func (p *DSPAParams) RetrieveOrCreateSecret(ctx context.Context, client client.Client, secretName, secretKey string, generatedPasswordLength int, log logr.Logger) (string, error) {
+	val, err := p.RetrieveSecret(ctx, client, secretName, secretKey, log)
+	if err != nil && apierrs.IsNotFound(err) {
+		generatedPass := passwordGen(generatedPasswordLength)
+		return base64.StdEncoding.EncodeToString([]byte(generatedPass)), nil
+	} else if err != nil {
+		log.Error(err, "Unable to create DB secret...")
+		return "", err
+	}
+	log.Info(fmt.Sprintf("Secret [%s] already exists, using stored value.", secretName))
+	return val, nil
+}
+
+func (p *DSPAParams) RetrieveOrCreateDBSecret(ctx context.Context, client client.Client, secret *dspa.SecretKeyValue, log logr.Logger) (string, error) {
+	dbPassword, err := p.RetrieveOrCreateSecret(ctx, client, secret.Name, secret.Key, config.GeneratedDBPasswordLength, log)
+	if err != nil {
+		return "", err
+	}
+	return dbPassword, nil
+
+}
+
+func (p *DSPAParams) RetrieveOrCreateObjectStoreSecret(ctx context.Context, client client.Client, secret *dspa.S3CredentialSecret, log logr.Logger) (string, string, error) {
+	accessKey, err := p.RetrieveOrCreateSecret(ctx, client, secret.SecretName, secret.AccessKey, config.GeneratedObjectStorageAccessKeyLength, log)
+	if err != nil {
+		return "", "", err
+	}
+	secretKey, err := p.RetrieveOrCreateSecret(ctx, client, secret.SecretName, secret.SecretKey, config.GeneratedObjectStorageSecretKeyLength, log)
+	if err != nil {
+		return "", "", err
+	}
+	return accessKey, secretKey, nil
+}
+
 // SetupDBParams Populates the DB connection Parameters.
 // If an external secret is specified, SetupDBParams will retrieve DB credentials from it.
 // If DSPO is managing a dynamically created secret, then SetupDBParams generates the creds.
 func (p *DSPAParams) SetupDBParams(ctx context.Context, dsp *dspa.DataSciencePipelinesApplication, client client.Client, log logr.Logger) error {
 
 	usingExternalDB := p.UsingExternalDB(dsp)
-
-	var customCreds *dspa.SecretKeyValue
-
-	// Even if a secret is specified DSPO will deploy its own secret owned by DSPO
-	p.DBConnection.CredentialsSecret = &dspa.SecretKeyValue{
-		Name: config.DBSecretNamePrefix + p.Name,
-		Key:  config.DBSecretKey,
-	}
-
 	if usingExternalDB {
 		// Assume validation for CR ensures these values exist
 		p.DBConnection.Host = dsp.Spec.Database.ExternalDB.Host
 		p.DBConnection.Port = dsp.Spec.Database.ExternalDB.Port
 		p.DBConnection.Username = dsp.Spec.Database.ExternalDB.Username
 		p.DBConnection.DBName = dsp.Spec.Database.ExternalDB.DBName
-		customCreds = dsp.Spec.Database.ExternalDB.PasswordSecret
+		p.DBConnection.CredentialsSecret = dsp.Spec.Database.ExternalDB.PasswordSecret
+
+		// Retreive DB Password from specified secret.  Handle errors (empty password) later
+		password, _ := p.RetrieveSecret(ctx, client, p.DBConnection.CredentialsSecret.Name, p.DBConnection.CredentialsSecret.Key, log)
```

Comment:
not sure how I feel about dropping the err like this, we should handle it properly if an error is encountered, if the resource is not found handle it when handling the error via `apierrs.IsNotFound(err)`

currently we could be hiding other forms of errors we may not necessarily be aware of, so we should expose them